homeassistant/components/seventeentrack/coordinator.py

```diff
@@ -0,0 +1,91 @@
+"""Coordinator for 17Track."""
+
+from dataclasses import dataclass
+from typing import Any
+
+from py17track import Client as SeventeenTrackClient
+from py17track.errors import SeventeenTrackError
+from py17track.package import Package
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from .const import (
+    CONF_SHOW_ARCHIVED,
+    CONF_SHOW_DELIVERED,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    LOGGER,
+)
+
+
+@dataclass
+class SeventeenTrackData:
+    """Class for handling the data retrieval."""
+
+    summary: dict[str, dict[str, Any]]
+    live_packages: dict[str, Package]
+
+
+class SeventeenTrackCoordinator(DataUpdateCoordinator[SeventeenTrackData]):
+    """Class to manage fetching 17Track data."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, client: SeventeenTrackClient) -> None:
+        """Initialize."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=DEFAULT_SCAN_INTERVAL,
+        )
+        self.show_delivered = self.config_entry.options[CONF_SHOW_DELIVERED]
+        self.account_id = client.profile.account_id
+
+        self._show_archived = self.config_entry.options[CONF_SHOW_ARCHIVED]
+        self._client = client
+
+    async def _async_update_data(
+        self,
+    ) -> SeventeenTrackData:
+        """Fetch data from 17Track API."""
+
+        summary = {}
+        summary_dict = {}
+        live_packages = set()
+        live_packages_dict = {}
+
+        try:
+            summary = await self._client.profile.summary(
+                show_archived=self._show_archived
+            )
+
+        except SeventeenTrackError as err:
+            LOGGER.error("There was an error retrieving the summary: %s", err)
+
+        for status, quantity in summary.items():
+            summary_dict[slugify(status)] = {
+                "quantity": quantity,
+                "packages": [],
+                "status_name": status,
+            }
+
+        try:
+            live_packages = set(
+                await self._client.profile.packages(show_archived=self._show_archived)
+            )
+        except SeventeenTrackError as err:
+            LOGGER.error("There was an error retrieving the packages: %s", err)
```

Comment:
Should we raise `UpdateFailed` when one of these 2 calls fail?

Reply:
I'm not sure what is the added value for raising it. If either has an error we can continue with empty data for one but the other will have data populated

Reply:
How stable is the API? If its not excessively flakey I would suggest just raising it

Reply:
OK. raising

Reply:
These 2 calls can now be merged

Reply:
merged. Thanks